e2e/container/run_test.go

```diff
@@ -34,6 +35,22 @@ func TestRunAttachedFromRemoteImageAndRemove(t *testing.T) {
 	golden.Assert(t, result.Stderr(), "run-attached-from-remote-and-remove.golden")
 }
 
+// Regression test for https://github.com/docker/cli/issues/5053
+func TestRunInvalidEntrypointWithAutoremove(t *testing.T) {
+	environment.SkipIfDaemonNotLinux(t)
+
+	result := make(chan *icmd.Result)
+	go func() {
+		result <- icmd.RunCommand("docker", "run", "--entrypoint=invalidcommand", "--rm", fixtures.AlpineImage, "param")
```

Comment:
Really nit; slightly leaning toward simplifying the test (not having the `--entrypoint` in there if it's not really needed for the test), e.g.

```suggestion
		result <- icmd.RunCommand("docker", "run", "--rm", fixtures.AlpineImage, "invalidcommand")
```

Reply:
Sure!